src/crypto.rs

```diff
@@ -113,6 +113,45 @@ fn verify_rsa(
     Ok(res.is_ok())
 }
 
+/// Compares the signature given with a re-computed signature using the public key for RSA.
+///
+/// `signature` is the signature part of a jwt (text after the second '.')
+///
+/// `signing_input` is base64(header) + "." + base64(claims)
+pub fn verify_rsa(
+    signature: &str,
+    signing_input: &str,
+    (n, e): (&[u8], &[u8]),
+    algorithm: Algorithm,
+) -> Result<bool> {
+    let maybe_rsa_parameters = match algorithm {
+        Algorithm::RS256 => Some(&signature::RSA_PKCS1_2048_8192_SHA256),
+        Algorithm::RS384 => Some(&signature::RSA_PKCS1_2048_8192_SHA384),
+        Algorithm::RS512 => Some(&signature::RSA_PKCS1_2048_8192_SHA512),
+        _ => None,
```

Comment:
actually, can probably `return` here so `rsa_parameters` isn't an `Option`

Reply:
Done removed two levels of indentation as a result. Thanks!